extra/nrnivmodl-core.in

```diff
@@ -1,104 +1,121 @@
         echo "  -V              Verbose: show commands executed by make"
-        echo "  -p <n_procs>    Make parallelism. Default: 4"
+        echo "  -p <n_procs>    Number of parallel builds (Default: $PARALLEL_BUILDS)"
         exit 0;;
     ?)
         exit 1;;
     esac
 done
 
+# consume an option
 shift $(($OPTIND - 1))
 
+# only one mod files directory is supported in neuron and coreneuron
 if [ $# -gt 1 ]; then
     echo "[ERROR] $APP_NAME expects at most one mod dir. See syntax: '$APP_NAME -h' "
     exit 1
 fi
 
-
-# If defined mods dir be in $1
+# if defined mods dir be in $1
 if [ $# -eq 1 ]; then
     # Note: due to bug #712 makefile wont handle mod dir with spaces, so we let it fail here
     params_MODS_PATH=$1
 fi
+
+# warn if no mod files provided
 if [ -d $params_MODS_PATH ]; then
-    # warn if no mod files provided
-    ls $params_MODS_PATH/*.mod || echo "WARNING: No mods found!"
+    ls $params_MODS_PATH/*.mod &>/dev/null || echo "WARNING: No mod files found, compiling default ones only!"
 else
     echo "FATAL: Invalid mods directory: '$params_MODS_PATH'"
     exit 1
 fi
 
-# Output options & populate make_params
-mkdir -p "$(dirname "$SAVE_FILE")"
-echo "# nrnivmodl-core options. Mods location: $params_MODS_PATH" > $SAVE_FILE
+# temporary directory where mod files will be copied
+temp_mod_dir=@CMAKE_HOST_SYSTEM_PROCESSOR@/core/mods
+mkdir -p $temp_mod_dir
 
+# copy mod files with include files. note that ${ROOT_DIR}/share
+# has inbuilt mod files and user provided mod files are in $params_MODS_PATH.
+set +e
+for mod_dir in ${ROOT_DIR}/share/modfile $params_MODS_PATH;
+do
+    # copy mod files and include files
+    files=`ls $mod_dir/*.mod $mod_dir/*.inc 2>/dev/null`
+    for f in $files;
+    do
+        # copy mod files only if it's changed (to avoid rebuild)
+        target_file_path=$temp_mod_dir/`basename $f`
+        if ! diff -q $f $target_file_path &>/dev/null;  then
+            cp $f $target_file_path
+        fi
+    done
+done
+set -e
 
-# Build params to make command (and save them to file)
-make_params=("ROOT=${ROOTDIR}")
+# use new mod files directory for compilation
+params_MODS_PATH=$temp_mod_dir
 
+# build params to make command
+make_params=("ROOT=${ROOT_DIR}")
 for param in $MAKE_OPTIONS; do
     var="params_${param}"
     if [ "${!var+x}" ]; then
         make_params+=("$param=${!var}")
-        echo "$var=\"${!var}\""
     fi
-done >> "$SAVE_FILE"
-
-# If -d (deploy) provided, call "make install"
-if [ "$params_DESTDIR" ]; then
-    make_params+=("install")
-fi
+done
 
+# run makefile
 echo "[INFO] Running make with params: ${make_params[@]}"
-make -j$PARALLELISM -f "${ROOTDIR}/share/coreneuron/nrnivmodl_core_makefile" "${make_params[@]}"
+make -j$PARALLEL_BUILDS -f "${ROOT_DIR}/share/coreneuron/nrnivmodl_core_makefile" "${make_params[@]}"
 
-echo "[INFO] mods built successfully."
+echo "[INFO] MOD files built successfully for CoreNEURON"
```

Comment:
Nitpick.. missing new line

Reply:
(unix conformant)  `echo` already adds newline and hence not added.